pkg/user/tx_client.go

```diff
@@ -516,6 +517,47 @@ func (client *TxClient) EstimateGas(ctx context.Context, msgs []sdktypes.Msg, op
 	return client.estimateGas(ctx, txBuilder)
 }
 
+// EstimateGasPriceAndUsage simulates the transaction, calculating the amount of gas that was consumed during execution. The final
+// result will be multiplied by gasMultiplier(that is set in TxClient).
+// Also returns the estimated gas price given the provided priority.
+func (client *TxClient) EstimateGasPriceAndUsage(
+	ctx context.Context,
+	msgs []sdktypes.Msg,
+	priority gasestimation.TxPriority,
+	opts ...TxOption,
+) (float64, uint64, error) {
```

Comment:
[nits] 
- `float64` and `uint64` don't really communicate what is being returned, please add named return values. 
- The godoc describes the return arguments in the reverse order from how they are returned.
- GasUsage != GasLimit. Can we please consolidate on one term?

```suggestion
// EstimateGasPriceAndUsage returns the gasPrice and gasUsage for a set of messages given a priority. 
// Note: The returned gasPrice = estiamtedGasPrice * gasMultiplier (if set in TxClient).
func (client *TxClient) EstimateGasPriceAndUsage(
	ctx context.Context,
	msgs []sdktypes.Msg,
	priority gasestimation.TxPriority,
	opts ...TxOption,
) (gasPrice float64, gasUsage uint64, error) {
```

Reply:
6dbd9fc95a9bbdac817a188c8d53688c0de9f060